pkg/schedule/coordinator.go

```diff
@@ -157,13 +161,24 @@ func (c *Coordinator) IsPendingRegion(region uint64) bool {
 // The function is exposed for test purpose.
 func (c *Coordinator) PatrolRegions() {
 	defer logutil.LogPanic()
-
 	defer c.wg.Done()
 	ticker := time.NewTicker(c.cluster.GetCheckerConfig().GetPatrolRegionInterval())
 	defer ticker.Stop()
 
+	regionChan := make(chan *core.RegionInfo, patrolRegionChanLen)
+	quit := make(chan bool)
+	var wg sync.WaitGroup
+	defer func() {
+		close(regionChan)
+		close(quit)
+		wg.Wait()
+	}()
+	workersCount := c.cluster.GetCheckerConfig().GetPatrolRegionConcurrency()
+	c.startPatrolRegionWorkers(workersCount, regionChan, quit, &wg)
```

Comment:
I think we could merge this part of the logic into a separate function.

Reply:
Move them to the outside of `PatrolRegions`?

Reply:
Yes, I think some of the code here and below can be merged and moved out. You can try to see if it works.